Irrecoverable panic caused by empty Linkable include

Panic traces below.

Seems to be caused by this:
https://github.com/sst/sst/blob/dev/pkg/project/completed.go#L127

---

I assume this happened while I was writing a `Linkable` and saved `sst.config.ts` with an empty `include` array. This seems to have saved a corrupt sst manifest to the aws store.

Something like:
```typescript
    const tables = new sst.Linkable("Tables", {
      properties: {
        Table1,
        Table2,
      },
      include: [],
    })
```

I'll go and poke around in aws and see if I can roll this back.

---

Other than that, I have tried:
* rolling back to an earlier working commit
* killing related processes
* reinstalling sst
* deleting the .sst directory
* restarting my computer
* `sst unlock`

Running `sst@3.17.23`


---

## Panics

on `sst dev`
```
panic: recovered from errgroup.Group: interface conversion: interface {} is nil, not string
        goroutine 92 [running]:
        runtime/debug.Stack()
                /opt/hostedtoolcache/go/1.25.3/x64/src/runtime/debug/stack.go:26 +0x64
        golang.org/x/sync/errgroup.(*Group).add.func1.1()
                /home/runner/go/pkg/mod/golang.org/x/sync@v0.15.0/errgroup/errgroup.go:117 +0x200
        panic({0x105725d60?, 0x140017b2780?})
                /opt/hostedtoolcache/go/1.25.3/x64/src/runtime/panic.go:783 +0x120
        github.com/sst/sst/v3/pkg/project.getCompletedEvent({0x105b8ebc8, 0x14000837770}, {0x140003959b0, 0x2c}, 0x14000e0b7e0)
                /home/runner/work/sst/sst/pkg/project/completed.go:127 +0xd40
        github.com/sst/sst/v3/pkg/project.(*Project).RunNext(0x14000e10200, {0x105b8ebc8, 0x14000837770}, 0x140009ee190)
                /home/runner/work/sst/sst/pkg/project/run.go:126 +0xac8
        github.com/sst/sst/v3/pkg/project.(*Project).Run(...)
                /home/runner/work/sst/sst/pkg/project/run.go:39
        github.com/sst/sst/v3/cmd/sst/mosaic/deployer.Start({0x105b8ebc8, 0x14000837770}, 0x14000e10200, 0x14000fa3e48)
                /home/runner/work/sst/sst/cmd/sst/mosaic/deployer/deployer.go:45 +0x3a4
        main.CmdMosaic.func10()
                /home/runner/work/sst/sst/cmd/sst/mosaic.go:222 +0x50
        golang.org/x/sync/errgroup.(*Group).add.func1()
                /home/runner/go/pkg/mod/golang.org/x/sync@v0.15.0/errgroup/errgroup.go:128 +0x7c
        created by golang.org/x/sync/errgroup.(*Group).add in goroutine 1
                /home/runner/go/pkg/mod/golang.org/x/sync@v0.15.0/errgroup/errgroup.go:96 +0x78


goroutine 1 [running]:
golang.org/x/sync/errgroup.(*Group).Wait(0x14000e24240)
        /home/runner/go/pkg/mod/golang.org/x/sync@v0.15.0/errgroup/errgroup.go:70 +0x74
main.CmdMosaic(0x14000e10080)
        /home/runner/work/sst/sst/cmd/sst/mosaic.go:353 +0x12a0
github.com/sst/sst/v3/cmd/sst/cli.(*Cli).Run(0x14000e10080)
        /home/runner/work/sst/sst/cmd/sst/cli/cli.go:90 +0x110
main.run()
        /home/runner/work/sst/sst/cmd/sst/main.go:131 +0x34c
main.main()
        /home/runner/work/sst/sst/cmd/sst/main.go:55 +0x4d8
```

on `sst remove`
```
panic: interface conversion: interface {} is nil, not string

goroutine 1 [running]:
github.com/sst/sst/v3/pkg/project.getCompletedEvent({0x1036d6bc8, 0x140008373b0}, {0x14000997140, 0x2c}, 0x14000ba3de0)
        /home/runner/work/sst/sst/pkg/project/completed.go:127 +0xd40
github.com/sst/sst/v3/pkg/project.(*Project).RunNext(0x14000e1c200, {0x1036d6bc8, 0x140008373b0}, 0x1400014b630)
        /home/runner/work/sst/sst/pkg/project/run.go:126 +0xac8
github.com/sst/sst/v3/pkg/project.(*Project).Run(...)
        /home/runner/work/sst/sst/pkg/project/run.go:39
main.CmdRemove(0x14000e1c080)
        /home/runner/work/sst/sst/cmd/sst/remove.go:47 +0x3ac
github.com/sst/sst/v3/cmd/sst/cli.(*Cli).Run(0x14000e1c080)
        /home/runner/work/sst/sst/cmd/sst/cli/cli.go:90 +0x110
main.run()
        /home/runner/work/sst/sst/cmd/sst/main.go:131 +0x34c
main.main()
        /home/runner/work/sst/sst/cmd/sst/main.go:55 +0x4d8
```

@wilfredjonathanjames Tried your advice, I have the same issue, but I don't see any `include` without a `type` on my temp.json - Is there anything else worth looking for to pinpoint the issue? 

and I also got this issue without using any `include: [],` on my sst infra code - Today I updated some dependencies on my monorepo and ran the `sst dev` command and got this.

my logs:
```
bun sst dev                                                                               
panic: recovered from errgroup.Group: interface conversion: interface {} is nil, not string
	goroutine 230 [running]:
	runtime/debug.Stack()
		/opt/hostedtoolcache/go/1.25.4/x64/src/runtime/debug/stack.go:26 +0x64
	golang.org/x/sync/errgroup.(*Group).add.func1.1()
		/home/runner/go/pkg/mod/golang.org/x/sync@v0.15.0/errgroup/errgroup.go:117 +0x200
	panic({0x105999d60?, 0x14003da9ad0?})
		/opt/hostedtoolcache/go/1.25.4/x64/src/runtime/panic.go:783 +0x120
	github.com/sst/sst/v3/pkg/project.getCompletedEvent({0x105e02c48, 0x14000b93130}, {0x14001e4c060, 0x2c}, 0x14000fd60e0)
		/home/runner/work/sst/sst/pkg/project/completed.go:95 +0xec4
	github.com/sst/sst/v3/pkg/project.(*Project).RunNext(0x14000b76100, {0x105e02c48, 0x14000b93130}, 0x140006423c0)
		/home/runner/work/sst/sst/pkg/project/run.go:127 +0xac8
	github.com/sst/sst/v3/pkg/project.(*Project).Run(...)
		/home/runner/work/sst/sst/pkg/project/run.go:40
	github.com/sst/sst/v3/cmd/sst/mosaic/deployer.Start({0x105e02c48, 0x14000b93130}, 0x14000b76100, 0x14000c6ad68)
		/home/runner/work/sst/sst/cmd/sst/mosaic/deployer/deployer.go:45 +0x3a4
	main.CmdMosaic.func10()
		/home/runner/work/sst/sst/cmd/sst/mosaic.go:225 +0x50
	golang.org/x/sync/errgroup.(*Group).add.func1()
		/home/runner/go/pkg/mod/golang.org/x/sync@v0.15.0/errgroup/errgroup.go:128 +0x7c
	created by golang.org/x/sync/errgroup.(*Group).add in goroutine 1
		/home/runner/go/pkg/mod/golang.org/x/sync@v0.15.0/errgroup/errgroup.go:96 +0x78
	

goroutine 1 [running]:
golang.org/x/sync/errgroup.(*Group).Wait(0x14000114f00)
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.15.0/errgroup/errgroup.go:70 +0x74
main.CmdMosaic(0x14000bb0280)
	/home/runner/work/sst/sst/cmd/sst/mosaic.go:356 +0x12a0
github.com/sst/sst/v3/cmd/sst/cli.(*Cli).Run(0x14000bb0280)
	/home/runner/work/sst/sst/cmd/sst/cli/cli.go:90 +0x110
main.run()
	/home/runner/work/sst/sst/cmd/sst/main.go:131 +0x34c
main.main()
	/home/runner/work/sst/sst/cmd/sst/main.go:55 +0x4d8
error: "sst" exited with code 1

```

--- a/pkg/project/completed.go
+++ b/pkg/project/completed.go
@@ -58,11 +58,16 @@ func getCompletedEvent(ctx context.Context, passphrase string, workdir *PulumiWo
 	complete.Resources = deployment.Resources
 
 	for _, resource := range complete.Resources {
-		outputs := parsePlaintext(resource.Outputs).(map[string]interface{})
+		outputs, ok := parsePlaintext(resource.Outputs).(map[string]interface{})
+		if !ok {
+			continue
+		}
 		if resource.URN.Type().Module().Package().Name() == "sst" {
 			if resource.Type == "sst:sst:Version" {
-				if outputs["target"] != nil && outputs["version"] != nil {
-					complete.Versions[outputs["target"].(string)] = int(outputs["version"].(float64))
+				target, targetOk := outputs["target"].(string)
+				version, versionOk := outputs["version"].(float64)
+				if targetOk && versionOk {
+					complete.Versions[target] = int(version)
 				}
 			}
 
@@ -91,16 +96,23 @@ func getCompletedEvent(ctx context.Context, passphrase string, workdir *PulumiWo
 		}
 
 		if match, ok := outputs["_tunnel"].(map[string]interface{}); ok {
+			ip, ipOk := match["ip"].(string)
+			username, usernameOk := match["username"].(string)
+			privateKey, privateKeyOk := match["privateKey"].(string)
+			if !ipOk || !usernameOk || !privateKeyOk {
+				continue
+			}
 			tunnel := Tunnel{
-				IP:         match["ip"].(string),
-				Username:   match["username"].(string),
-				PrivateKey: match["privateKey"].(string),
+				IP:         ip,
+				Username:   username,
+				PrivateKey: privateKey,
 				Subnets:    []string{},
 			}
-			subnets, ok := match["subnets"].([]interface{})
-			if ok {
+			if subnets, ok := match["subnets"].([]interface{}); ok {
 				for _, subnet := range subnets {
-					tunnel.Subnets = append(tunnel.Subnets, subnet.(string))
+					if s, ok := subnet.(string); ok {
+						tunnel.Subnets = append(tunnel.Subnets, s)
+					}
 				}
 				complete.Tunnels[resource.URN.Name()] = tunnel
 			}
@@ -110,36 +122,43 @@ func getCompletedEvent(ctx context.Context, passphrase string, workdir *PulumiWo
 			complete.Hints[string(resource.URN)] = hint
 		}
 
-		if resource.Type == "sst:sst:LinkRef" && outputs["target"] != nil && outputs["properties"] != nil {
-			properties, ok := outputs["properties"].(map[string]interface{})
-			if !ok {
+		if resource.Type == "sst:sst:LinkRef" {
+			target, targetOk := outputs["target"].(string)
+			properties, propertiesOk := outputs["properties"].(map[string]interface{})
+			if !targetOk || !propertiesOk {
 				continue
 			}
 			link := common.Link{
 				Properties: properties,
 				Include:    []common.LinkInclude{},
 			}
-			if outputs["include"] != nil {
-				include, ok := outputs["include"].([]interface{})
-				if ok {
-					for _, include := range include {
-						link.Include = append(link.Include, common.LinkInclude{
-							Type:  include.(map[string]interface{})["type"].(string),
-							Other: include.(map[string]interface{}),
-						})
+			if includeSlice, ok := outputs["include"].([]interface{}); ok {
+				for _, inc := range includeSlice {
+					incMap, ok := inc.(map[string]interface{})
+					if !ok {
+						continue
 					}
+					incType, ok := incMap["type"].(string)
+					if !ok {
+						continue
+					}
+					link.Include = append(link.Include, common.LinkInclude{
+						Type:  incType,
+						Other: incMap,
+					})
 				}
 			}
-			complete.Links[outputs["target"].(string)] = link
+			complete.Links[target] = link
 		}
 	}
 
-	outputs := parsePlaintext(deployment.Resources[0].Outputs).(map[string]interface{})
-	for key, value := range outputs {
-		if strings.HasPrefix(key, "_") {
-			continue
+	if outputs, ok := parsePlaintext(deployment.Resources[0].Outputs).(map[string]interface{}); ok {
+		for key, value := range outputs {
+			if strings.HasPrefix(key, "_") {
+				continue
+			}
+			complete.Outputs[key] = value
 		}
-		complete.Outputs[key] = value
 	}
 
 	return complete, nil

